### PR TITLE
DB validations for currentCollector field in generator model

### PR DESCRIPTION
**Pull request by pallavi2209**

### CODE REVIEW

File: db/dbErrors.js [hunk 1]
```diff
@@ -139,6 +139,15 @@ errors.create({
   name: 'ReferencedByGenerator',
   parent: this.ValidationError,
 });
+errors.create({
+  scope: exports,
+  code: 10122,
+  status: 400,
+  name: 'InvalidCurrentCollector',
```
Comment (iamigo) [inline db/dbErrors.js]: InvalidCollector

File: db/dbErrors.js [hunk 2]
```diff
@@ -139,6 +139,15 @@ errors.create({
...
+  parent: this.ValidationError,
+  defaultMessage: 'CurrentCollector should be one of the assigned list of ' +
```
Comment (iamigo) [inline db/dbErrors.js]: The specified collector is not in the list of collectors for this sample generator.

File: db/helpers/generatorUtil.js [hunk 3]
```diff
@@ -131,9 +131,56 @@ function validateCollectors(seq, collectorNames,
...
+ * InvalidCurrentCollector error
+ */
+function validateCurrentCollector(currCollector, generatorCollectorsArr) {
+  if (!currCollector) {
+    return; // no need to validate
+  }
+
+  if (!generatorCollectorsArr || generatorCollectorsArr.length === 0) {
+    throw new dbErrors.InvalidCurrentCollector({
+      explanation: 'Assigned collector list of generator is empty.',
```
Comment (iamigo) [inline db/helpers/generatorUtil.js]: InvalidCollector

This sample generator has not yet designated any collectors.

File: db/helpers/generatorUtil.js [hunk 4]
```diff
@@ -131,9 +131,56 @@ function validateCollectors(seq, collectorNames,
...
+
+    genCollectorsNameArr.push(genCollectorName);
+
+    if (currCollector === genCollectorName) {
+      isCurrCollectorValid = true; // valid if names matches
+    }
+  });
+
+  if (!isCurrCollectorValid) {
+    throw new dbErrors.InvalidCurrentCollector({
```
Comment (iamigo) [inline db/helpers/generatorUtil.js]: InvalidCollector

Collector "${...}" cannot be assigned as the current collector for this sample generator. Select one of ${...}.

**Comment by pallavi2209** [comment]
@shriramshankar updated the PR

**Pull request merged**